### DISCUSSION TITLE
Import forms in typescript

### DISCUSSION THREAD

**Discussion by ThomasThurnher** [danielo515/obsidian-modal-form#348] [Q&A]
**Is your feature request related to a problem? Please describe.**

I would like to import the data for a new form from a JSON file. I would like to use this data very much like the interaction in typescript with the "openForm" functionality, that is instead of :

```
const result = await modalForm.openForm({
				title: "Choose TimelineProfile",
				fields: [
					{
						...
					  }
				],
			});
 ```

I would like to interact with it as (or some workaround of that)

```
const result = await modalForm.openForm({
				"pathTo/myFormData.json"
			});
 ```

**Describe the solution you'd like**

Either, I would like to input the data via a string which is parsed from JSON, or via direct input of the JSON file.

Is this possible somehow? Thank you for your answer and I greatly appreciate your work.

**Comment by danielo515** [comment]
Interestingly I wanted to segregate the form data from the plugin settings for a long time now. I have to confirm it, but I think that once you remove a plugin from obsidian, it's settings are also removed, and that is very bad for users. The reason I didn't done that yet is because it introduces more complexity that I don't have time to deal with now.
That said, this is not yet possible with modal form alone, but thanks to the flexibility that having minimal APIs provide (sorry for praising myself here) this is easily achievable with some extra code:

```ts
const definition = await tp.file.include("[[path/yourFile.json]]").then(x => JSON.parse(x));
const result = await modalForm.openForm(definition)
```

**Comment by ThomasThurnher** [comment]
@danielo515  Thank you very much for your feedback - this solution works for me!

**Comment by danielo515** [comment]
> @danielo515 Thank you very much for your feedback - this solution works for me!

Glad to see it does!
